ui/components/app/simulation-details/fiat-display.tsx

```diff
@@ -0,0 +1,80 @@
+import React from 'react';
+import { useSelector } from 'react-redux';
+import {
+  TextColor,
+  TextVariant,
+} from '../../../helpers/constants/design-system';
+import { useI18nContext } from '../../../hooks/useI18nContext';
+import { Text } from '../../component-library';
+import { getCurrentLocale } from '../../../ducks/locale/locale';
+import { getCurrentCurrency } from '../../../selectors';
+import { SizeNumber } from '../../component-library/box/box.types';
+import { BalanceChange, FIAT_UNAVAILABLE, FiatAmountAvailable } from './types';
+
+const textStyle = {
+  color: TextColor.textAlternative,
+  variant: TextVariant.bodySm,
+  paddingRight: 2 as SizeNumber,
+};
+
+const FiatNotAvailableDisplay: React.FC = () => {
+  const t = useI18nContext();
+  return <Text {...textStyle}>{t('simulationPreviewFiatNotAvailable')}</Text>;
+};
+
+/**
+ * Returns a function that formats a fiat amount as a localized string.
+ */
+const useFiatFormatter = () => {
```

Comment:
This is so nice, I'm surprised it doesn't already exist.

How are we currently rendering fiat values in confirmations for example? Just this but not via a hook?